### ISSUE TITLE
Item modifications weren't downloaded 

### ISSUE THREAD

**Issue by iruthaya** [stephanrauh/ngx-extended-pdf-viewer#1592]
I've added [showEditor] ="true" to my code, but when I download the file, none of the changes I made are reflected there. 

**One more clarification**

How can we Save modified PDF file with out downloading the file?

kindly assist me

**Comment by stephanrauh** [comment]
What? Oh. That's bad news. The last time I tried, the file I downloaded included the changes. Currently I'm drowning in new, difficult issues, but still, this seems to be an issue I should put at the top of the pile.

Please brace yourself for my questions. At the moment, I'm short of time, so I appreciate every help you can give me.

Just doublechecking: if you modify a PDF file on the Mozilla showcase, does it include your changes?

**Comment by iruthaya** [comment]
I appreciate your prompt response. 

if you modify a PDF file on the Mozilla showcase, does it include your changes?
Yes, it contains my modifications, but they are not reflected when using this. 

Is it possible to Save modified PDF with out downloading the file?

**Comment by stephanrauh** [comment]
If it works on Mozilla's showcase, it's probably a bug I've added.

> Is it possible to Save modified PDF with out downloading the file?

What's the difference? Maybe "download" is a confusing word. Everything happens on the client, so it's not really a download.

**Comment by iruthaya** [comment]
Thank you for your response. I'm using this in the IONIC application, and I need to add some annotations and remarks before sending the file back to the server without without downloading and re-uploading. Is that possible?

**Comment by stephanrauh** [comment]
I see. Yes, this sounds like a valid use-case, one of the use-cases many other developers might benefit from.

I believe it's possible to implement such a feature. Basically, it's stripping down the download functionality. I haven't tried it yet, but I suppose all that remains is this code (or something similar):
```ts
      const PDFViewerApplication: IPDFViewerApplication = (window as any).PDFViewerApplication;
      PDFViewerApplication._ensureDownloadComplete();

      const file = await PDFViewerApplication.pdfDocument.saveDocument();
     // that's the file you're interested in
```

These days, I'm short of time, so I encourage you to try the code snippet and tell me how it went. I'm pretty sure I've missed a bit or two, but if you start debugging the `DownloadManager` class in the JavaScript (not TypeScript!) code, you'll get the knack of it quickly.

**Comment by stephanrauh** [comment]
Your bug fix has landed with version 16.0.0-alpha.1. The programatic download feature is still on my list.

For some reason unknown, my copy of Chrome refuses to print now. I don't think that's got anything to do with the bugfix, but you never know.

**Comment by stephanrauh** [comment]
It seems the Chrome problem was a temporary glitch.
